nucliadb/src/nucliadb/search/search/query.py

```diff
@@ -372,8 +376,22 @@ async def parse_vector_search(self, request: nodereader_pb2.SearchRequest) -> bo
 
     async def parse_relation_search(self, request: nodereader_pb2.SearchRequest) -> list[str]:
         autofilters = []
+        # BUG: autofiler should autofilter, not enable relation search
         if self.has_relations_search or self.autofilter:
-            detected_entities = await self.fetcher.get_detected_entities()
+            if self.query_entities:
+                detected_entities = []
+                for entity in self.query_entities:
+                    relation_node = utils_pb2.RelationNode()
+                    relation_node.value = entity.name
+                    if entity.type is not None:
+                        relation_node.ntype = utils_pb2.RelationNode.NodeType.Value(
```

Comment:
Can we use the models we have for this?

Change the KnowledgeGraphEntity.type to be nucliadb_models.metadata.RelationNodeType

and here use the conversion map `RelationNodeTypeMap` to convert to the protobuffer type.

Reply:
Sure